### ISSUE TITLE
Consider disabling experimental Rust syntax

### ISSUE THREAD

**Issue by mikupls** [spikecodes/libreddit#303]
Hi! I'm absolutely new to Rust, and the dev tooling setup was a bit confusing (I'm very surprised the official website tells you to run `curl|sh`...). I ended up instead installing it via apt on Debian (testing). It seems like Rust absolutely forbids using "experimental" syntax using the "stable" tooling, and so I'm unable to build:

```
error[E0658]: or-patterns syntax is experimental
   --> src/main.rs:263:8
    |
263 |         Some("best" | "hot" | "new" | "top" | "rising" | "controversial") => subreddit::community(req).boxed(),
    |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #54883 <https://github.com/rust-lang/rust/issues/54883> for more information

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
```

This is a selfish request, because I could also just install Rust nightly. However, since this is the only use of exp. syntax, I believe rewriting this somehow without exp. syntax could be an okay trade-off vs. the increased ease of getting started for beginners like me.

I have absolutely no clue how this line could be rewritten; maybe it is difficult; or maybe this request is dumb. So feel free to close in these cases :)

PS: Currently my fix is to simply comment that line out when building.

**Comment by SuperSamus** [comment]
(I'm not a maintainer)
Or patterns were stabilized in Rust 1.53 (latest stable right now is 1.56). I suggest that you install Rust from Rustup, because Debian is Debian and they aren't known for having the latest packages ([it seems it's at version 1.50](https://packages.debian.org/testing/rustc), which is [from February](https://blog.rust-lang.org/2021/02/11/Rust-1.50.0.html)).

**Comment by mikupls** [comment]
Thanks @SuperSamus for the reply. I was not aware that the rust feature had been merged to stable in the meantime, and it is only Debian that is lagging behind so much. In that case, I will close this issue.

**Comment by mikupls** [comment]
Aaand Debian testing just merged 1.56, and compilation works now :tada:
